src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -244,9 +244,12 @@ internal static IQueryMatch BuildQuery(Parameters builderParameters, out OrderMe
                 var maxTermToScan = builderParameters.Take switch
                 {
                     < 0 => int.MaxValue, // meaning, take all
+                    // We cannot apply this optimization when we are returning statistics (RavenDB-21525).
+                    var take when builderParameters.Query.SkipStatistics => take + 1, 
                     int.MaxValue => int.MaxValue, // avoid overflow
-                    var take => take + 1
+                    _ => int.MaxValue
```

Comment:
Opps, we need to make the `int.MaxValue` be _higher_ than the `var take`, otherwise we risk overflowign.

Reply:
Pushed to `long` to avoid overflowing.